src/main/java/org/folio/service/transactions/cancel/CancelPendingPaymentService.java

```diff
@@ -9,32 +10,43 @@
 import javax.money.CurrencyUnit;
 import javax.money.Monetary;
 import java.util.List;
-import java.util.Map;
-
-import static org.apache.commons.collections4.CollectionUtils.isNotEmpty;
 
 public class CancelPendingPaymentService extends CancelTransactionService {
 
-  public CancelPendingPaymentService(BudgetService budgetService) {
-    super(budgetService);
+  public CancelPendingPaymentService(BudgetService budgetService, TransactionDAO paymentCreditDAO, TransactionDAO encumbranceDAO) {
+    super(budgetService, paymentCreditDAO, encumbranceDAO);
+  }
+
+  @Override
+  Budget budgetMoneyBack(Budget budget, List<Transaction> transactions) {
+    Budget newBudget = JsonObject.mapFrom(budget).mapTo(Budget.class);
+    CurrencyUnit currency = Monetary.getCurrency(transactions.get(0).getCurrency());
+    transactions.forEach(tmpTransaction -> {
+      double newAwaitingPayment = MoneyUtils.subtractMoney(newBudget.getAwaitingPayment(),
+        tmpTransaction.getAmount(), currency);
+      double newVoidedAmount = tmpTransaction.getAmount();
+
+      newBudget.setAwaitingPayment(newAwaitingPayment);
+      tmpTransaction.setVoidedAmount(newVoidedAmount);
+      tmpTransaction.setAmount(0.0);
+    });
+    return newBudget;
+  }
+
+  @Override
+  String getEncumbranceId(Transaction pendingPayment) {
```

Comment:
Please replace return value with Optional<String> to better match the new Java style